demo/tool/update_status.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:polymer_elements/paper_item.dart';
+import 'package:polymer_elements/paper_toolbar.dart';
+import 'package:polymer_elements/iron_ajax.dart';
+import 'package:polymer_elements/paper_styles.dart';
+
+/// Silence analyzer [UpdateStatus], [IronAjax], [PaperIconButton], [PaperItem],
+/// [PaperDrawerPanel], [PaperHeaderPanel], [PaperMenu], [IronList],
+/// [PaperToolbar], [PaperTooltip],
+@PolymerRegister('update-status')
+class UpdateStatus extends PolymerElement {
+  UpdateStatus.created() : super.created();
+
+  @property Map data;
+
+  @Property(
+      computed: 'getOrganizations(data)', observer: 'organizationsChanged')
+  List<String> organizations;
+
+  @property String selectedOrganization;
+
+  @Property(
+      computed: 'getRepositories(data, selectedOrganization, repositoryFilter)',
+      observer: 'repositoriesChanged')
+  List<Map> repositories;
+
+  @property Object selectedRepository;
+
+  @Property(
+      computed:
+          'getRepositoryData(data, selectedOrganization, selectedRepository)')
+  Map repositoryData;
+
+  @property Map selectedFile;
+
+  @property String repositoryFilter = 'any-changes';
+
+  @reflectable
+  void organizationsChanged(newValue, oldValue) {
+    final IronList list = $['organizations'];
+    if (newValue == null || newValue.isEmpty) {
+      list.clearSelection();
+    } else {
+      list.selectItem(newValue.first);
+    }
+  }
+
+  @reflectable
+  bool isRepositorySelected() => selectedOrganization != null;
+
+  @reflectable
+  void repositoriesChanged(newValue, oldValue) {
+    new Future(() {
+      final IronList list = $$('#repositories');
+      if (newValue == null || newValue.isEmpty) {
+        list.clearSelection();
+      } else {
+        list.selectItem(newValue.first);
+      }
+    });
+  }
+
+  @reflectable
+  List<String> getOrganizations(Map data) {
+    if (data == null) {
+      return [];
+    }
+    return data['organizations'].map((org) => org['organization']).toList();
+  }
+
+  @reflectable
+  List<Map> getRepositories(
+      Map data, String organization, String repositoryFilter) {
+    if (data == null || organization == null) {
+      return [];
+    }
+    Iterable repositories = data['organizations'].firstWhere(
+        (org) => org['organization'] == organization)['repositories'];
+    if (repositoryFilter != null &&
+        repositoryFilter.isNotEmpty &&
+        repositoryFilter != 'all' &&
+        repositoryFilter != 'any-changes') {
+      repositories = repositories.where(
+          (repo) => repo['change'] == 'RepositoryChange.${repositoryFilter}');
+    } else if (repositoryFilter == 'any-changes') {
+      repositories = repositories.where((repo) =>
+          repo['change'] != 'RepositoryChange.none' ||
+              (repo['files'] as List)
+                  .where((file) => file['change'] != 'FileChange.none')
+                  .isNotEmpty);
+    }
+    return repositories.map((repo) => convertToJs(repo)).toList();
```

Comment:
you shouldn't have to manually use `convertToJs` manually, if you do then its probably a bug